x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -657,6 +661,94 @@ public void getUserPrivileges(AuthorizationInfo authorizationInfo, ActionListene
         }
     }
 
+    @Override
+    public void getRemoteAccessRoleDescriptorsIntersection(
+        final String remoteClusterAlias,
+        final AuthorizationInfo authorizationInfo,
+        final ActionListener<RoleDescriptorsIntersection> listener
+    ) {
+        if (authorizationInfo instanceof RBACAuthorizationInfo == false) {
+            listener.onFailure(
+                new IllegalArgumentException("unsupported authorization info:" + authorizationInfo.getClass().getSimpleName())
+            );
+            return;
+        }
+
+        final Role role = ((RBACAuthorizationInfo) authorizationInfo).getRole();
+        final RemoteIndicesPermission remoteIndicesPermission;
+        try {
+            remoteIndicesPermission = role.remoteIndices().forCluster(remoteClusterAlias);
+        } catch (UnsupportedOperationException e) {
+            // TODO we will need to implement this to support API keys with assigned role descriptors
+            listener.onFailure(
+                new IllegalArgumentException(
+                    "cannot retrieve remote access role descriptors for API keys with assigned role descriptors.",
+                    e
+                )
+            );
+            return;
+        }
+
+        if (remoteIndicesPermission.remoteIndicesGroups().isEmpty()) {
+            listener.onResponse(RoleDescriptorsIntersection.EMPTY);
+            return;
+        }
+
+        final List<RoleDescriptor.IndicesPrivileges> indicesPrivileges = new ArrayList<>();
+        for (RemoteIndicesPermission.RemoteIndicesGroup remoteIndicesGroup : remoteIndicesPermission.remoteIndicesGroups()) {
+            for (IndicesPermission.Group indicesGroup : remoteIndicesGroup.indicesPermissionGroups()) {
```

Comment:
I think we settled on there being no requirement (neither short, nor medium term) for remote privileges other than index, so I would prefer to punt on this for now. 

FWIW, I don't see an immediate issue with multiple different remote privilege types; `remote_applications` could simply map to `applications`, same as `remote_indices` maps to `indices`. One way or another, an authenticated subject always maps to exactly one Role instance (possibly built from multiple role descriptors), and we will need to capture all remote privileges (indices or otherwise) in one or multiple role descriptors here. I don't think we should go down this rabbit hole right now though.

Reply:
Agreed. No short or medium term requirement for other remote privileges. Collapsing multiple `role_descriptor.remote_indices` into a single `role_descriptor.indices` is sufficient for now.

I recall `remote_run_as` was discussed and excluded, so no need to over engineer this right now. If that or other remote privileges are added later, we should come back to this code and change it. I think we would need a one-to-one mapping of `role_descriptor.remote_indices` to `role_descriptor.indices`, and introduce canonical ordering `role_descriptor`. IBackwards/forwards compatibility is not an issue, because this is a singleton set with room to grow later.